activerecord/lib/active_record/attribute_methods/write.rb

```diff
@@ -36,8 +36,7 @@ def write_attribute(attr_name, value)
         end
 
         name = self.class.primary_key if name == "id".freeze && self.class.primary_key
```

Comment:
Looking at this here, I also wonder what the implications would be of making `id` an "ordinary" attribute_alias.(But that's not related to this PR.)